include/DecisionTree.hpp

```diff
@@ -112,7 +117,7 @@ class DecisionTree {
     void setHowManyTotalTrainingSamples(int howManyTotalTrainingSamples);
     void setRootNode(std::unique_ptr<DecisionTreeNode> rootNode);
 
-  private:
+  protected:
```

Comment:
since strictly used by a derived or matching class, protected allows Evaluate to access/modify member variables without complex setters